mutiny-core/src/onchain.rs

```diff
@@ -107,10 +107,23 @@ impl<S: MutinyStorage> OnChainWallet<S> {
                 5,
             )
             .await?;
-        // get new wallet lock for writing
-        let mut wallet = self.wallet.try_write()?;
-        wallet.apply_update(update)?;
-        wallet.commit()?;
+
+        // get new wallet lock for writing and apply the update
+        match self.wallet.try_write() {
+            Ok(mut wallet) => match wallet.apply_update(update) {
+                Ok(_) => wallet.commit()?,
```

Comment:
should `commit` be matched too?

Reply:
No, that's just writing to storage which will return a proper error